### PR TITLE
Add exception case for the `translate` function

### PR DESCRIPTION
**Pull request by yariks5s**
related: https://github.com/ClickHouse/ClickHouse/pull/71441

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Provides exception in case the second argument is shorter than the third for the `translate` function, so now queries like `SELECT translate('aAbBcC', 'ab', 'abc')` throw an exception.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit c197a28babd64192c2f31d285864fb871b30a705 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/72893/c197a28babd64192c2f31d285864fb871b30a705/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/72893/c197a28babd64192c2f31d285864fb871b30a705/style_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: src/Functions/translate.cpp [hunk 1]
```diff
@@ -44,14 +44,6 @@ struct TranslateImpl
...
         }
 
-        // Handle any remaining characters in map_from by assigning a default value
-        for (size_t i = min_size; i < map_from.size(); ++i)
-        {
-            if (!isASCII(map_from[i]))
-                throw Exception(ErrorCodes::BAD_ARGUMENTS, "Second argument must be ASCII strings");
-            map[static_cast<unsigned char>(map_from[i])] = ascii_upper_bound + 1;
-        }
-
```
Comment (evillique) [inline src/Functions/translate.cpp:54]: Why do we need to remove this part? It seems like we would need to remove the next `for` loop instead

Comment (yariks5s) [reply src/Functions/translate.cpp:54]: yes, I will change it

**Comment by rschu1ze** [comment]
(no need to provide a changelog entry for PRs of category "Not for changelog" :) )

**Review by evillique — APPROVED** [review]

**Comment by yariks5s** [comment]
AST Fuzzer found a bug here: 
```SELECT json, tuple(tuple((tuple('!,', toUInt128(2), 2, toLowCardinality(2), 2, materialize(2), 2, 2, 2, 2), '123', toLowCardinality(3), 3, toNullable(toLowCardinality(3)), 3, 3))), json.d.a.:`Array(Nullable(Int64))`, json.d.b.:Int64, json.d.b.:Date FROM test ORDER BY tuple(tuple('123', 3, toLowCardinality(3), 3, 3, 3, 3)) DESC NULLS LAST, toLowCardinality('!,') DESC NULLS LAST, json DESC NULLS FIRST, id ASC NULLS LAST FORMAT JSONColumns```

seem to be not related

**Comment by rschu1ze** [comment]
@yariks5s Please update the changelog entry. It currently doesn't even mention which function was changed. If possible, also include a brief example of what didn't work before and what works now.

**Comment by clickgapai** [comment]
Hi @yariks5s @evillique — while reviewing this PR I found the following:

- **Bug (P3):** `translateUTF8` error message reports byte sizes instead of code point counts — https://github.com/ClickHouse/ClickHouse/issues/104786

Happy to discuss — close anything that's wrong or already addressed.

**Pull request merged**